code/components/ShippingCheckoutComponent.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+class ShippingCheckoutComponent extends CheckoutComponent{
+
+	public function getFormFields(Order $order) {
+		$fields = new FieldList();
+		$estimates = $order->getShippingEstimates();
+		
+		$fields->push(
+			OptionsetField::create(
+				"ShippingMethodID",
+				"Shipping Options",
+				$estimates->map(),
+				$estimates->First()->ID
+			)
+		);
+
+		return $fields;
+	}
+
+	public function getRequiredFields(Order $order) {
+		
+		return array();
+		
+	}
+
+	public function validateData(Order $order, array $data) {
+		$result = new ValidationResult();
+		if(!isset($data['ShippingMethodID'])){
+			$result->error("Shipping method not provided", "ShippingMethod");
+			throw new ValidationException($result);
+		}		
+		
+		if(!ShippingMethod::get()->byID($data['ShippingMethodID'])){
+		 	$result->error("Gateway not supported", "ShippingMethod");
```

Comment:
Ah, no it should be 'Shipping Method does not exist' instead of 'gateway not supported'